src/strategies/masterchef-pool-balance-indexed/index.ts

```diff
@@ -0,0 +1,117 @@
+export const version = '0.0.1';
+
+/*
+ * Generic masterchef pool balance strategy. Accepted options:
+ * - chefAddress: masterchef contract address
+ * - pid: mastechef pool id (starting with zero)
+ * - uniPairAddress: address of a uniswap pair (or a sushi pair or any other with the same interface)
+ *    - if the uniPairAddress option is provided, converts staked LP token balance to base token balance
+ *      (based on the pair total supply and base token reserve)
+ *    - if uniPairAddress is null or undefined, returns staked token balance as is
+ * - tokenIndex: index of a token in LP pair, optional, by default 0
+ * - weight: integer multiplier of the result (for combining strategies with different weights, totally optional)
+ */
+
+const abi = [
+  'function userInfo(uint256, address) view returns (uint256 amount, uint256 rewardDebt)',
+  'function totalSupply() view returns (uint256)',
+  'function getReserves() view returns (uint112 _reserve0, uint112 _reserve1, uint32 _blockTimestampLast)',
+  'function token0() view returns (address)',
+  'function token1() view returns (address)',
+  'function decimals() view returns (uint8)'
+];
+
+const indexAbi = ['function index() public view returns (uint256)'];
+
+// calls is a 1-dimensional array so we just push 3 calls for every address
+const getCalls = (addresses: any[], options: any) => {
+  const result: any[] = [];
+  for (const address of addresses) {
+    result.push([options.chefAddress, 'userInfo', [options.pid, address]]);
+    if (options.uniPairAddress != null) {
+      result.push([options.uniPairAddress, 'totalSupply', []]);
+      result.push([options.uniPairAddress, 'getReserves', []]);
+    }
+  }
+  return result;
+};
+
+function arrayChunk<T>(arr: T[], chunkSize: number): T[][] {
+  const result: T[][] = [];
+  for (let i = 0, j = arr.length; i < j; i += chunkSize) {
+    result.push(arr.slice(i, i + chunkSize));
+  }
+  return result;
+}
+
+// values is an array of (chunked) call results for every input address
+// for setups with uniPairAddress each chunk has 3 items, for setups without, only 1 item
+function processValues(values: any[], options: any, index: number): number {
+  const poolStaked = values[0][0];
+  const weight: BigNumber = BigNumber.from(options.weight || 1);
+  const weightDecimals: BigNumber = BigNumber.from(10).pow(
+    BigNumber.from(options.weightDecimals || 0)
+  );
+  let result: BigNumber;
+  if (!options.uniPairAddress) {
+    result = poolStaked.mul(weight).div(weightDecimals);
+  } else {
+    const uniTotalSupply = values[1][0];
+    const uniReserve = values[2][options.tokenIndex || 0];
+    const precision = BigNumber.from(10).pow(18);
+    const tokensPerLp = uniReserve.mul(precision).div(uniTotalSupply);
+    result = poolStaked
+      .mul(tokensPerLp)
+      .mul(weight)
+      .div(weightDecimals)
+      .div(precision);
+  }
+  return (
+    parseFloat(formatUnits(result.toString(), options.decimals || 18)) * index
```

Comment:
I think contract-call + math (multiplier) should work here as well, please try and let me know :)

Reply:
sorry, i'm not tracking here. Happy to adjust, just need some further clarification. To keep things consistent with the existing masterchef-pool-balance strategy we're returning the value the same way as it is already implemented on that strategy, just now multiplying the return value by the index.

Reply:
I think what you're referring to is using the math strategy, and using contract call for the index variable and the masterchef-pool-balance with a multiplier, instead of creating this strategy. 

The challenge is that this assumes the indexed token address is on the same chain as the LP. This usually isn't the case... and we need the ability to retrieve the index from a specific network (the indexNetwork param in this strategy, for us its mainnet), plus the balance on whatever network is specified by the strategy during setup.